Client/Frontend/Browser/UserScriptManager.swift

```diff
@@ -202,4 +234,23 @@ class UserScriptManager {
       }
     }
   }
+  
+  @MainActor func injectSolanaInternalScript(tab: Tab, solanaInternalScript: String?) async {
+    guard let webView = tab.webView,
+          let solanaInternalScript = solanaInternalScript else {
+      return
+    }
+    // inject the internal solana script
+    let script = """
+    window.__firefox__.execute(function($, $Object) {
+      \(solanaInternalScript)
```

Comment:
Where is solanaInternalScript coming from here? This looks like it would allow remote JS execution

Reply:
This script comes from BraveCore. This `injectSolanaInternalScript(tab:solanaInternalScript:)` function is called from the [SolanaProviderScriptHandler](https://github.com/brave/brave-ios/blob/solana-dapps/Client/Frontend/UserContent/UserScripts/Scripts_Dynamic/ScriptHandlers/Paged/SolanaProviderScriptHandler.swift#L165) with the value set on each Tab in [TabManager.swift](https://github.com/brave/brave-ios/blob/solana-dapps/Client/Frontend/Browser/TabManager.swift#L509). The script is retrieved from BraveCore in a closure in [BrowserViewController.swift](https://github.com/brave/brave-ios/blob/solana-dapps/Client/Frontend/Browser/BrowserViewController.swift#L427). 

The solana internal script itself is this [internal script](https://github.com/brave/brave-core/blob/1.45.x/components/brave_wallet/resources/solana_provider_internal.js), which is being [removed in the next release (v1.46)](https://github.com/brave/brave-core/pull/15247).

Reply:
@StephenHeaps we don't inject this if the wallet is locked, so a webpage can overwrite `_brave_solana.createPublickey` and `_brave_solana.createTransaction`. I'm not sure about the security implications, but we freeze the object here:
https://github.com/brave/brave-core/blob/803186400aa12f0d29016ad2f95b958c0f880eda/components/brave_wallet/resources/solana_provider_internal.js#L28
probably for a good reason. Is there a reason we don't inject at the document's start?

Reply:
Core team advised to lazy-load this script due to performance concerns when loading the Solana Web3 library, some additional info in the [brave-core issue](https://github.com/brave/brave-browser/issues/22468).

Reply:
Ok, so I checked on Desktop, and there is the same issue. I'll ask Anthony (can't mention him here for some reason) in the PR you linked

Reply:
Ok, so in this case it sounds like the script is trusted, but we've still got a few issues in terms of the object getting overwritten. In the case of `createPublicKey` that could present issues with the account being returned causing issues. I don't see a large impact here, but this is highly contextual to the dApp in question so it certainly could create a larger impact in the future.

In the case of `createTransaction` this seems like it could present a legitimate concern to modify a transaction before it's signed such that the user could be tricked into signing it later when believing it came as a legitimate dApp request. In this case, I can see legitimate harm coming from this if it were able to be overwritten while the wallet is locked and I think this should get fixed. It seems @stoletheminerals suggestion of injecting at the start seems like a safe bet, but given there's a performance impact here that's going to have to be worked around as well.

Reply:
v1.46 of core removes the `window._brave_solana` property entirely in favour of injecting the minified js solanaWeb3 library (I have a [wip branch available here against 1.46](https://github.com/brave/brave-ios/compare/solana-dapps...solana-dapps-solanaWeb3Library#diff-6d62f27259cf433b454bdcd9bc43c8f83a8345487158bf3d635015164226cefeR245)). I _believe_ this change was made to address the mentioned issue, but we'll have to investigate if this is possible to bring in v1.45 of iOS without an uplift of that change.

Reply:
@StephenHeaps, are we still going to lazy-load the minified solanaWeb3 lib? If so, I'm afraid the issue could persist.
@kdenhartog, do you have any attack vectors in mind for `createTransaction`?

Reply:
Yeah the plan was to continue to lazy-load the minified solanaWeb3 library on 1.46 for the performance concerns (cc @darkdh). The current changes are calling the constructors directly through `solanaWeb3` instead of through `createPublickey` / `createTransaction` functions.

Reply:
The requiring webpacked solanaWeb3.js code will be evaluated on every page load because we use it when defining properties. We used to load it on start and get flagged by automatic performance monitoring (https://github.com/brave/brave-browser/issues/22468).
Now we don't pollute the global namespace with `_brave_solana` which means the website has no access to where we construct public key or transaction. Desktop and Android are doing that in v8 and I suggested iOS construct those in a way that website won't be able to interfere. It is to prevent various attack vectors toward  `window._brave_solana` while our SafeBuiltins can only protect it from prototype polluted. (Here are some examples of how to do that https://bravesoftware.slack.com/archives/C0414R6NUKT/p1663952287601889?thread_ts=1663886636.872799&cid=C0414R6NUKT).
Let me know if we want brave-core's [`window._brave_solana` removal](https://github.com/brave/brave-core/pull/15247) in 1.45 so iOS can release Solana dapps with it.

Reply:
^^ As @darkdh suggested, we should load in a way that `_brave_solana` is not globally accessible

Reply:
Thanks @darkdh.
The iOS code changes for removing `_brave_solana` from global namespace are available here:
https://github.com/brave/brave-ios/compare/solana-dapps...solana-dapps-solanaWeb3Library
This branch was tested using a local build of `brave-core`, so will likely fail to build or run without a local build against `brave-core`'s `master` branch, but at least code can be viewed.